src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -0,0 +1,735 @@
+            mappedImportRow.setGermplasm( germplasmPIO );
+
+            if (! StringUtils.isBlank( importRow.getGid() )) { // if GID is blank, don't bother to check if it is valid.
+                validateGermplasm(importRow,validationErrors, i, germplasmPIO);
+            }
+            // Construct Observations -- Done in another card
+            mappedBrAPIImport.put(i, mappedImportRow);
+        }
+        // End-of-loop
+
+        validationErrors = validateFields(importRows, validationErrors);
+
+        if (validationErrors.hasErrors() ){
+            throw new ValidatorException(validationErrors);
+        }
+
+        // Construct our response object
+        return getStatisticsMap(importRows);
+    }
+
+    private void getNewBrapiData(List<BrAPIImport> importRows, Program program, boolean commit) {
+
+        String opsUnitSequenceName = program.getObsUnitSequence();
+        if (opsUnitSequenceName == null) {
+            log.error(String.format("Program, %s, is missing a value in the obsUnit sequence column.", program.getName()));
+            throw new HttpStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "Program is not properly configured for observation unit import");
+        }
+        Supplier<BigInteger> obsUnitNextVal = () -> dsl.nextval(opsUnitSequenceName.toLowerCase());
+
+        // It is assumed, at this point, that there is only one experiment per import-file
+        String expSeqValue = null;
+        if (commit) {
+            String expUnitSequenceName = program.getExpSequence();
+            if (expUnitSequenceName == null) {
+                log.error(String.format("Program, %s, is missing a value in the exp sequence column.", program.getName()));
+                throw new HttpStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "Program is not properly configured for experiment import");
+            }
+            expSeqValue = dsl.nextval(expUnitSequenceName.toLowerCase()).toString();
+        }
+
+        for (BrAPIImport row : importRows) {
+            ExperimentObservation importRow = (ExperimentObservation) row;
+
+            PendingImportObject<BrAPITrial> trialPIO = createTrialPIO(program, commit, importRow);
+            this.trialByNameNoScope.put(importRow.getExpTitle(), trialPIO);
+
+            PendingImportObject<BrAPILocation> locationPIO = createLocationPIO(importRow);
+            this.locationByName.put(importRow.getEnvLocation(), locationPIO);
+
+            PendingImportObject<BrAPIStudy> studyPIO = createStudyPIO(program, commit, expSeqValue, importRow);
+            this.studyByNameNoScope.put(importRow.getEnv(), studyPIO);
+
+            PendingImportObject<BrAPIObservationUnit> obsUnitPIO = createObsUnitPIO(program, commit, obsUnitNextVal, importRow);
+            String key = createOUKey(importRow);
+            this.observationUnitByNameNoScope.put(key, obsUnitPIO);
+        }
+    }
+
+    private String createOUKey(ExperimentObservation importRow) {
```

Comment:
Might be useful to add comment saying OU stands for Observation Unit?

Reply:
Will change the name of the method to createObservationUnitKey().  Thanks